openhands/controller/agent_controller.py

```diff
@@ -449,7 +450,21 @@ async def _step(self) -> None:
                 == ActionConfirmationStatus.AWAITING_CONFIRMATION
             ):
                 await self.set_agent_state_to(AgentState.AWAITING_USER_CONFIRMATION)
-            self.event_stream.add_event(action, EventSource.AGENT)
+
+            if isinstance(action, CmdRunAction):
+                # Split the command into multiple CmdRunAction instances
+                commands = split_bash_commands(action.command)
+                for i, cmd in enumerate(commands):
+                    if not cmd:
+                        continue
+                    new_action = CmdRunAction(command=cmd)
+                    if i < len(commands) - 1:
+                        new_action.thought = ''
```

Comment:
what's this about?

Reply:
This was from the original prompt:
```.. Each instance should have a blank 'thought' attribute, except for the last one which should have the thought from the original action...```

The reason for this is that it prevents the frontend UI from printing the thought multiple times. e.g.: (Without this functionality):
![image](https://github.com/user-attachments/assets/f09e5f96-ab70-4112-afbd-7d598a9b3833)